src/components/ApplicationDashboard.js

```diff
@@ -317,33 +352,81 @@ const Dashboard = ({
+            <Checkbox
+              checked={safewalk}
+              onChange={handleChange}
+              label="If you are planning to walk home alone on campus on the night of the 21st, would you like organizers to accompany you to your destination?"
             />
-            <Button onClick={() => setDietaryRestrictions(dietaryInput)}>Save</Button>
-          </DietaryNoteContainer>
-          <RSVPButton
-            width="flex"
-            onClick={isRsvpOpen && (() => setRSVP(canRSVP))}
-            shouldDisplay={canRSVP || hackerStatus === 'acceptedAndAttending'}
-            color={canRSVP ? 'primary' : 'secondary'}
-            disabled={!isRsvpOpen}
-          >
-            {canRSVP ? 'RSVP' : 'un-RSVP'}
-          </RSVPButton>
+          </SafeWalkContainer>
+        )}
+
+        <FooterContainer>
+          {/* Hides this option if a user unRSVP'd */}
+          {hackerRSVPStatus !== "Un-RSVP'd" && (
+            <DietaryNoteContainer>
+              <TextInput
+                value={dietaryInput}
+                onChange={e => setDietaryInput(e.target.value)}
+                placeholder="Dietary Restrictions/Notes"
+                color="primary"
+              />
+              <Button onClick={() => setDietaryRestrictions(dietaryInput)}>Save</Button>
+            </DietaryNoteContainer>
+          )}
+
+          {/* Only show button if a user hasn't unRSVPed yet and can still RSVP*/}
+          {hackerRSVPStatus !== "Un-RSVP'd" && canRSVP && (
+            <RSVPButton
+              width="flex"
+              onClick={isRsvpOpen && (() => setRSVP(canRSVP))}
+              shouldDisplay={canRSVP || hackerStatus === 'acceptedAndAttending'}
+              color={canRSVP ? 'primary' : 'secondary'}
+              disabled={!isRsvpOpen}
+            >
+              RSVP
+            </RSVPButton>
+          )}
+
+          {/* If the user can unRSVP, pop up the placeholder button which pops up a modal */}
+          {hackerRSVPStatus !== "Un-RSVP'd" && !canRSVP && (
+            <>
+              <Button
+                width="flex"
+                color={canRSVP ? 'primary' : 'secondary'}
+                onClick={() => setdisplayUnRSVPModel('block')}
+              >
+                un-RSVP
+              </Button>
+
+              {displayUnRSVPModel === 'block' && (
+                <UnRSVPModelContainer>
+                  <UnRSVPModelTint onClick={() => setdisplayUnRSVPModel('none')} />
+                  <UnRSVPModel>
+                    <p style={{ marginTop: '30px', marginLeft: '10px' }}>
+                      Are you sure that you want to un-RSVP? You won’t be able to RSVP again.
+                    </p>
+
+                    <Button width="flex" onClick={() => setdisplayUnRSVPModel('none')}>
+                      Cancel
+                    </Button>
+                    <RSVPButton
+                      width="flex"
+                      onClick={isRsvpOpen && (() => setRSVP(canRSVP))}
```

Comment:
I'm trying to un-rsvp from this event, but `isRsvpOpen` is false from the feature flag so this isn't working @naijwu @DonaldKLee